analyzers/tests/SonarAnalyzer.UnitTest/TestCases/InfiniteRecursion.CSharp11.cs

```diff
@@ -14,6 +14,68 @@ interface InfiniteRecursion
 
     public class Addition : IAdditionOperators<Addition, Addition, Addition>
     {
-        public static Addition operator +(Addition left, Addition right) => left + right; // FN
+        public static Addition operator +(Addition left, Addition right) => left + right; // Noncompliant
+//                                      ^
+    }
+
+    public class Comparison : IComparisonOperators<Comparison, Comparison, Comparison>
+    {
+        public static Comparison operator ==(Comparison? left, Comparison? right)
+//                                        ^^
+        {
+            return left == right;
+        }
+
+        public static Comparison operator !=(Comparison left, Comparison right) => left == right; // Compliant (we do not support cross-method analysis)
+
+        public static Comparison operator <(Comparison left, Comparison right) =>
+            left is null ? left < right : left; // Compliant
```

Comment:
Note: this particular case is actually an FN (because it's still an infinite loop for the case when `left` is null) 

But this is the case for the existing logic

e.g.
```
public int Foo(int i)
{
  if (i < 0) return Foo(i); // FN
  return i;
}
```

So no action, I just want to document this on the PR